### PR TITLE
progress_circular

### PR DESCRIPTION
**Pull request by jaroslavrehorka**
Hi,
I've implemented basic functionality of progress_circular. I was following Angular Material and only thing I did differently is that I'm not hiding preloader when there is empty mode, but I use [hidden].

### CODE REVIEW

File: examples/components/progress_circular/basic_usage.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  <p>For operations where the user is asked to wait a moment while something finishes up, and it's not necessary to
+     expose what's happening behind the scenes and how long it will take, use an indeterminate indicator.</p>
+  <div layout="row" layout-sm="column" layout-align="space-around">
+    <md-progress-circular mode="indeterminate"></md-progress-circular>
+  </div>
+
+  <h2 class="md-title">Indeterminate</h2>
+
+  <p>
+    Your current theme colors can be used to easily colorize your progress indicator with `md-warn` or `md-accent`
```
Comment (justindujardin) [inline examples/components/progress_circular/basic_usage.html]: The examples don't support markdown, perhaps wrap the `md-warn` and `md-accent` notes in `<code></code>` blocks?

**Comment by justindujardin** [comment]
Hey, great! Thanks for contributing. :+1: 

I'd like to have a basic spec for new components before merging, can you add one? If not I'll add one later and merge.  

Thanks!

**Comment by jaroslavrehorka** [comment]
I will try to add one later today. Thank you for your great work!

**Comment by justindujardin** [comment]
I look forward to it. I added a spec for the progress_linear component earlier today if you would like to look at it for reference.

**Comment by jaroslavrehorka** [comment]
Thanks for example!

Tests are done! I've added some functionality testing used by angular material team in their tests.

There is one more thing I did differently - they manipulate with `md-mode` attribute a lot - if there is no `md-mode` they automatically init it's value to indeterminate, I did that using SASS (`:not([mode]`).

**Comment by justindujardin** [comment]
@jaroslavrehorka thanks a bunch, it looks really nice!  I will publish it tomorrow with some other changes.

**Comment by justindujardin** [comment]
Published with v0.1.0
https://justindujardin.github.io/ng2-material/#/components/progress-circular

:+1:

**Pull request merged**